From b59ceaf9729765ff08dafd83dfbcb8da4772f413 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@cavium.com>
Date: Thu, 23 Feb 2012 12:58:42 -0800
Subject: [PATCH 215/236] NET: octeon-ethernet: Their are 4 SRIO ports in OCTEON II.

Source: Cavium SDK 2.3-427

ebb63xx board has 2 SRIO interfaces, each has 2 ports, port 40 - 43.
ebb66xx has one SRIO interface with 4 ports, port 40 - 43.

Signed-off-by: Chandrakala Chavva <cchavva@cavium.com>
Integrated-by: Jiang Bin <bin.jiang@windriver.com>
---
 drivers/net/octeon/ethernet-napi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/octeon/ethernet-napi.c b/drivers/net/octeon/ethernet-napi.c
index 64946ef..db173ee 100644
--- a/drivers/net/octeon/ethernet-napi.c
+++ b/drivers/net/octeon/ethernet-napi.c
@@ -335,7 +335,7 @@ static int CVM_OCT_NAPI_POLL(struct napi_struct *napi, int budget)
 				priv = cvm_oct_by_pkind[port];
 		} else {
                         /* srio priv is based on mbox, not port */
-			if (port == 40 || port == 41) {
+			if (port >= 40 && port <= 43) {
 				const cvmx_srio_rx_message_header_t *rx_header =
 					(const cvmx_srio_rx_message_header_t *)skb->data;
 				priv = cvm_oct_by_srio_mbox[(port - 40) >> 1][rx_header->word0.s.mbox];
-- 
1.7.0

